### ISSUE TITLE
big loss of training speed performance with latest caffe version

### ISSUE THREAD

**Issue by engharat** [NVIDIA/caffe#215]
Hi,
having seen some bugs solved in the latest version, I switched from nvcaffe 15.8 to 15.10. I think this is the only thing that I've changed; after the usual compile with cmake (cudnn 5.1,cuda 7.5, drivers 361.62, NCCL, 2 titan X cards) , I've run some usual training experiments con object recognition with a caffenet, training it from scratch: 100 epochs went from 7 days to finish (DIGITS estimate) to 13 days finish. It is not an estimation error; the training is REALLY slowed by 50%. Plus, because of some CPU calculations on a custom layer, I usually got the GPU usage throttling from 80% to 2%, meaning that I was almost CPU-limited in training performances.
Now my GPU is running at 60-80% all the time, meaning I'm GPU limited. it seems to me that the gpu speed has greatly decreased.
I'll try to downgrade back to 15.8: anyway, does anyone notice that performance degradation?
P.S. the nvidia-smi command info on performances (nvidia-smi -q -d PERFORMANCE) showed me that the GPUs where, during the training with 15.8 caffe, at P2 state, caused by "unknown".
 Now they are in P0 state, caused by "Applications Clocks Setting" or  "SW Power Cap"(throttling between those two answers)

**Comment by drnikolaev** [comment]
@engharat could you please try 0.15.9? Meanwhile, we are working on this issue.

**Comment by engharat** [comment]
I' m getting the same low performance with 0.15.9 and I finally get the usual performance going back to 0.15.8!

**Comment by drnikolaev** [comment]
@engharat Thanks! Could you do us a favour to check https://github.com/drnikolaev/caffe/tree/caffe-0.15.10-fix please?

**Comment by engharat** [comment]
@drnikolaev  sure! Unfortunately the fix https://github.com/drnikolaev/caffe/tree/caffe-0.15.10-fix  didn't worked; still getting halved speed.

**Comment by drnikolaev** [comment]
@engharat thank you. We have strong reasons to believe that we fixed it now, testing is in progress. To get early access to the fix please re-fetch  https://github.com/drnikolaev/caffe/tree/caffe-0.15.10-fix  and re-try it again.

**Comment by engharat** [comment]
@drnikolaev  the fix works! congratulations and thanks :)

**Comment by engharat** [comment]
@drnikolaev 
While my CNN training is running via DIGITS, I test the current snapshot of the network using the FC7 features on another task, executing on python:


```
   caffe.set_device(GPU_ID)
    caffe.set_mode_gpu()

# load a new model
return caffe.Net(deploy_file, caffemodel, caffe.TEST)
```

I have 2 TITAN X cards, so GPU 0 and GPU 1. When GPU 0 ram was half empty, the above code runs fine on GPU 0.  Now that I'm using all GPU 0 ram for a big CNN, i'm trying to run this code on GPU 1. But I get back this error:
`WARNING: Logging before InitGoogleLogging() is written to STDERR
E0824 18:02:47.545102  7896 common.cpp:116] Cannot create cuda stream. Cuda stream won't be available.
E0824 18:02:47.755434  7896 common.cpp:121] Cannot create Cublas handle. Cublas won't be available.
F0824 18:02:47.755472  7896 common.cpp:123] Check failed: status == CUBLAS_STATUS_SUCCESS (1 vs. 0)  CUBLAS_STATUS_NOT_INITIALIZED
*** Check failure stack trace: ***
`

In few words:  the system is failing in using the GPU #1 even if is half empty when GPU #0 has no memory available.I have always switched between GPU #0 and #1 for those types of tasks with no problems. 

Edit: it seems to me that is happening this:
https://github.com/BVLC/caffe/issues/440
Edit: I can confirm that happens only on the 0.15.10-fix, reverting back to 0.15.08 fixed the problem. So the 0.15.10-fix version introduced this bug (or maybe is an unsolved bug from 0.15.10 or 0.15.9)
